include/exadg/structure/material/library/st_venant_kirchhoff.cpp

```diff
@@ -180,41 +204,75 @@ StVenantKirchhoff<dim, Number>::second_piola_kirchhoff_stress_symmetrize(
 {
-  if(large_deformation)
-  {
-    return (this->second_piola_kirchhoff_stress_symmetrize(
-      get_E<dim, Number>(get_F<dim, Number>(gradient_displacement)), cell, q));
-  }
-  else
-  {
-    return (this->second_piola_kirchhoff_stress_symmetrize(gradient_displacement, cell, q));
-  }
+  return (this->second_piola_kirchhoff_stress_symmetrize(
+    get_E<dim, Number>(get_F<dim, Number>(gradient_displacement)), cell, q));
+}
+
+template<int dim, typename Number>
+dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
+StVenantKirchhoffSmallDeformation<dim, Number>::second_piola_kirchhoff_stress(
+  tensor const &     gradient_displacement,
+  unsigned int const cell,
+  unsigned int const q) const
+{
+  return (this->second_piola_kirchhoff_stress_symmetrize(gradient_displacement, cell, q));
 }
 
 template<int dim, typename Number>
 dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-StVenantKirchhoff<dim, Number>::second_piola_kirchhoff_stress_displacement_derivative(
-  dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & gradient_increment,
-  dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & deformation_gradient,
-  unsigned int const                                              cell,
-  unsigned int const                                              q) const
+StVenantKirchhoffLargeDeformation<dim, Number>::
+  second_piola_kirchhoff_stress_displacement_derivative(tensor const &     gradient_increment,
+                                                        tensor const &     deformation_gradient,
+                                                        unsigned int const cell,
+                                                        unsigned int const q) const
 {
   // Exploit linear stress-strain relationship and symmetrizing in
   // second_piola_kirchhoff_stress_symmetrize
   return (this->second_piola_kirchhoff_stress_symmetrize(
     transpose(deformation_gradient) * gradient_increment, cell, q));
 }
 
-template class StVenantKirchhoff<2, float>;
-template class StVenantKirchhoff<2, double>;
+template<int dim, typename Number>
+dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
+StVenantKirchhoffSmallDeformation<dim, Number>::
+  second_piola_kirchhoff_stress_displacement_derivative(tensor const &     gradient_increment,
+                                                        tensor const &     deformation_gradient,
+                                                        unsigned int const cell,
+                                                        unsigned int const q) const
+{
+  (void)deformation_gradient;
+
+  AssertThrow(false,
+              dealii::ExcMessage(
+                "This code path is not implemented in NonLinearOperator::do_cell_integral ."));
+
+  // Exploit linear stress-strain relationship and symmetrizing in
+  // second_piola_kirchhoff_stress_symmetrize
+  return (this->second_piola_kirchhoff_stress_symmetrize(gradient_increment, cell, q));
+}
```

Comment:
I think this function actually shows that inheritance might not be the method of choice to combine linear/nonlinear behavior. Inheritance is appropriate if one has a true "is-a" relationship, but a `StVenantKirchhoffSmallDeformation` does not seem to "be a" `StVenantKirchhoffBase` as this function shows.